### PR TITLE
Add more E2E to test different scenarios

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

Ensure we don't break any functionality in the future.

In this PR, new E2E tests from 1.0 have been ported to 2.0:
* Checking same principal is given by alternative origins.
* Setting the delegation expiration.
* Edge cases of handling post messages from the application.

# Changes

* New test on alternativeOrigins.spec.ts
* New test files delegationTtl.spec.ts and postMessage.spec.ts

### CODE REVIEW

File: src/frontend/tests/e2e-playwright/authorize/alternativeOrigins.spec.ts [hunk 1]
```diff
@@ -203,3 +203,79 @@ test("Should not follow redirect returned by /.well-known/ii-alternative-origins
...
+
+  // Use existing passkey
+  auth(authPage2);
+  await authPage2.getByRole("button", { name: "Primary account" }).click();
+
+  // Wait for II window to close
+  await authPage2.waitForEvent("close");
+
+  // Get the second principal
+  const principal2 = await page.locator("#principal").textContent();
```
Comment (sea-snake) [inline src/frontend/tests/e2e-playwright/authorize/alternativeOrigins.spec.ts:278]: Are we sure we won't have a possible flaky issue here? I mean it could grab the text content just before it's updated instead of after right?

Maybe the following waits (with a timeout) till the text matches?

```ts
await page.locator("#principal").toHaveText(principal1);
```

Comment (sea-snake) [reply src/frontend/tests/e2e-playwright/authorize/alternativeOrigins.spec.ts:278]: Edit:

It seems like we're not doing this in the `authorize` util either.

The Playwright docs seems to explicitly mention the risk of flakiness: https://playwright.dev/docs/api/class-locator#locator-text-content

Comment (lmuntaner) [reply src/frontend/tests/e2e-playwright/authorize/alternativeOrigins.spec.ts:278]: I'll add `await expect(page.locator("#principal")).not.toBeEmpty();` before getting the content.

I also added it to the authorize util

**Review by sea-snake — APPROVED** [review]
